internal-services/catalog/create-advisory-task.yaml

```diff
@@ -0,0 +1,102 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: task-create-advisory
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+      Pushes an advisory yaml to a Git repository
+  params:
+    - name: advisory_json
+      type: string
+      description: |
+          String containing a JSON representation of the advisory data. It should not contain the `advisory`
+          top level key
+          E.g. '{"spec":{"product_id":123,"type":"RHSA"}}'
+    - name: application
+      type: string
+      description: Application being released
+    - name: origin
+      type: string
+      description: |
+          The origin workspace where the release CR comes from.
+          This is used to determine the advisory path
+  results:
+    - name: result
+      description: Success if the task succeeds, the error otherwise
+  steps:
+    - name: create-advisory
+      image: quay.io/hacbs-release/release-utils:518c0423a3b3502c7b9ef6fd8652995bec2b4e1a
+      env:
+        - name: GITLAB_HOST
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: gitlab_host
+        - name: ACCESS_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: gitlab_access_token
+        - name: GIT_AUTHOR_NAME
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: git_author_name
+        - name: GIT_AUTHOR_EMAIL
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: git_author_email
+      script: |
+          #!/usr/bin/env sh
+          set -eox pipefail
+
+          REPO_BRANCH=main
+
+          # loading git and gitlab functions
+          . /home/utils/gitlab-functions
+          . /home/utils/git-functions
+          gitlab_init
+          git_functions_init
+
+          function get_advisory_num {
+              # This function will skip many advisory numbers as it is so tied to date. Need to
+              # discover if skipping advisory numbers like this is okay.
+              # There is also a race condition picking the advisory number like this.
+              EXISTING=$(for advisory in $(find data/advisories/*/${1} -name "advisory.yaml") ; do
+                  yq '.metadata.name' $advisory | grep ":${2}" || true
+              done | sort -r | head -n 1)
+              if [[ ! -z "$EXISTING" ]] ; then
+                  echo "${2}$((${EXISTING:9}+1))" # Strip first 9 chars for YYYY:MMDD then add 1
+                  return
+              fi
+              echo "${2}0" # First advisory of the day
+          }
+
+          # This also cds into the git repo
+          git_clone_and_checkout --repository "$(jq -rc '.advisory.repo' $(params.advisory_json))" \
```

Comment:
According to the description of the `advisory_json` pipeline parameter, this won't have the `advisory` key at the top level. So I believe this should be `jq -rc '.repo'`.

Reply:
Ah, yes, good catch

Reply:
I will test it one more time once I have https://github.com/redhat-appstudio/release-service-catalog/pull/245 passing CI before I merge here, but still open for review

Reply:
Tested with latest code here and on https://github.com/redhat-appstudio/release-service-catalog/pull/245 and it works as expected (including the failure scenario)